sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -173,8 +173,17 @@ private ShouldRetryResult shouldRetryOnSessionNotAvailable() {
             retryDelay = Duration.ofMillis(ClientRetryPolicy.RetryIntervalInMS);
         }
         this.retryContext = new RetryContext(this.failoverRetryCount, false);
-        return this.globalEndpointManager.refreshLocationAsync(null, forceRefresh)
-                .then(Mono.just(ShouldRetryResult.retryAfter(retryDelay)));
+        Mono<Void> completable = this.globalEndpointManager.refreshLocationAsync(null, forceRefresh);
+        if (isReadRequest || WebExceptionUtility.isWebExceptionRetriable(e)) {
```

Comment:
HttpConstants.StatusCodes.FORBIDDEN, HttpConstants.SubStatusCodes.FORBIDDEN_WRITEFORBIDDEN will qualify as isWebExceptionRetriable ? If not then how does that case handled ?

Reply:
I don't think it will be qualified as `WebExceptionRetriable`
And in that case, we force refresh but I don't think we will retry. 
Worth discussing over a meeting.